rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java

```diff
@@ -71,6 +72,15 @@ public long getMaxNumber() {
         return ByteUtil.byteArrayToLong(metadata.get(MAX_BLOCK_NUMBER_KEY));
     }
 
+    @Override
+    public long getMinNumber() {
+        if (index.isEmpty()) {
+            return -1;
```

Comment:
I personally don't like returning special values on special situations. What about having a `BlocksIndex#isEmpty` and make this method throw something like `IllegalStateException` if the internal index is empty? or even just throw the exception and not provide the other method if the special situation doesn't need to be handled

Reply:
Yup! You are right, I got distracted by the already existing logic. I decided to return an optional. Empty if the blockstore has no min number.